torch/onnx/symbolic_helper.py

```diff
@@ -477,18 +477,40 @@ def _interpolate_warning(interpolate_mode):
                   "We recommend using opset 11 and above for models using this operator.")
 
 def _unsqueeze_helper(g, input, axes_i):
-    if _export_onnx_opset_version >= 13:
-        axes = g.op("Constant", value_t=torch.tensor(axes_i, dtype=torch.long))
-        return g.op("Unsqueeze", input, axes)
-    else:
-        return g.op("Unsqueeze", input, axes_i=axes_i)
+    if _is_constant(axes_i[0]):
+        if _export_onnx_opset_version >= 13:
+            axes = g.op("Constant", value_t=torch.tensor(axes_i, dtype=torch.long))
+            return g.op("Unsqueeze", input, axes)
+        else:
+            return g.op("Unsqueeze", input, axes_i=axes_i)
+    else:  # Tensor type
+        if _export_onnx_opset_version >= 13:
```

Comment:
Handling errors first, and then remove else block. This makes the code a bit easier to read IMO: all error handling happens up front, so we know the invariants when we get to the rest of the code. Also decreases indentation.

```python
if _export_onnx_opset_version < 13:
  raise ValueError("Opset version must be >= 13 for Unsqueeze with dynamic axes.")
return g.op("Unsqueeze", input, axes_i[0])
```